common/CommonAPI.h

```diff
@@ -806,6 +369,53 @@ class CommonAPI
+		SFeatureRequest<nbl::video::IAPIConnection::E_FEATURE> optionalInstanceFeatures = {};
+		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> requiredDeviceFeatures = {};
+		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> optionalDeviceFeatures = {};
+		
+		nbl::asset::IImage::E_USAGE_FLAGS swapchainImageUsage = nbl::asset::IImage::E_USAGE_FLAGS::EUF_COLOR_ATTACHMENT_BIT;
+
+		constexpr static inline std::array<nbl::asset::E_FORMAT, 4> defaultAcceptableSurfaceFormats = { nbl::asset::EF_R8G8B8A8_SRGB, nbl::asset::EF_R8G8B8A8_UNORM, nbl::asset::EF_B8G8R8A8_SRGB, nbl::asset::EF_B8G8R8A8_UNORM };
+		constexpr static inline std::array<nbl::asset::E_COLOR_PRIMARIES, 1> defaultAcceptableColorPrimaries = { nbl::asset::ECP_SRGB };
+		constexpr static inline std::array<nbl::asset::ELECTRO_OPTICAL_TRANSFER_FUNCTION, 1> defaultAcceptableEotfs = { nbl::asset::EOTF_sRGB };
+		constexpr static inline std::array<nbl::video::ISurface::E_PRESENT_MODE, 1> defaultAcceptablePresentModes = { nbl::video::ISurface::EPM_FIFO_RELAXED };
+		constexpr static inline std::array<nbl::video::ISurface::E_SURFACE_TRANSFORM_FLAGS, 2> defaultAcceptableSurfaceTransforms = { nbl::video::ISurface::EST_IDENTITY_BIT, nbl::video::ISurface::EST_HORIZONTAL_MIRROR_ROTATE_180_BIT };
+
+		const nbl::asset::E_FORMAT* acceptableSurfaceFormats = &defaultAcceptableSurfaceFormats[0];
+		uint32_t acceptableSurfaceFormatCount = defaultAcceptableSurfaceFormats.size();
+		const nbl::asset::E_COLOR_PRIMARIES* acceptableColorPrimaries = &defaultAcceptableColorPrimaries[0];
+		uint32_t acceptableColorPrimaryCount = defaultAcceptableColorPrimaries.size();
+		const nbl::asset::ELECTRO_OPTICAL_TRANSFER_FUNCTION* acceptableEotfs = &defaultAcceptableEotfs[0];
+		uint32_t acceptableEotfCount = defaultAcceptableEotfs.size();
+		const nbl::video::ISurface::E_PRESENT_MODE* acceptablePresentModes = &defaultAcceptablePresentModes[0];
+		uint32_t acceptablePresentModeCount = defaultAcceptablePresentModes.size();
+		const nbl::video::ISurface::E_SURFACE_TRANSFORM_FLAGS* acceptableSurfaceTransforms = &defaultAcceptableSurfaceTransforms[0];
+		uint32_t acceptableSurfaceTransformCount = defaultAcceptableSurfaceTransforms.size();
+
+		nbl::asset::E_FORMAT depthFormat = nbl::asset::EF_UNKNOWN;
+
+		nbl::core::smart_refctd_ptr<nbl::ui::IWindow> window = nullptr;
+		nbl::core::smart_refctd_ptr<nbl::ui::IWindowManager> windowManager = nullptr;
+		nbl::core::smart_refctd_ptr<CommonAPIEventCallback> windowCb = nullptr;
+		nbl::core::bitflag<nbl::system::ILogger::E_LOG_LEVEL> logLevel =
+			nbl::core::bitflag(nbl::system::ILogger::ELL_DEBUG) | nbl::system::ILogger::ELL_PERFORMANCE | nbl::system::ILogger::ELL_WARNING | nbl::system::ILogger::ELL_ERROR | nbl::system::ILogger::ELL_INFO;
+
+		constexpr bool isHeadlessCompute()
+		{
+			return swapchainImageUsage == nbl::asset::IImage::E_USAGE_FLAGS(0);
+		}
+	};
+
 	struct InitOutput
 	{
 		enum E_QUEUE_TYPE
```

Comment:
Let's move these outside, CommonAPI::EQT_GRAPHICS looks much better than CommonAPI::InitOutput::EQT_GRAPHICS

Reply:
its not about looks, its about encapsulation, does anyone else exceptthings that deal with `InitOutput` need to deal with this enum, if not, it should stay in the deepest scope possible

Reply:
kind of, this gets used every time an example wants to fetch a queue

Reply:
Yea places that happen much later than CommonAPI::Init has happened. you can see things like queues[CommonAPI::InitOutput::EQT_GRAPHICS]